opcua/gopcua/subscribe.go

```diff
@@ -145,32 +154,43 @@ func (c client) runHandler(sub *opcuaGopcua.Subscription, cfg opcua.Config) erro
 			switch x := res.Value.(type) {
 			case *uaGopcua.DataChangeNotification:
 				for _, item := range x.MonitoredItems {
-					msg := opcua.Message{
-						ServerURI: cfg.ServerURI,
-						NodeID:    cfg.NodeID,
+					msg := message{
+						ServerURI: uri,
+						NodeID:    node,
 						Type:      item.Value.Value.Type().String(),
 						Time:      item.Value.SourceTimestamp.Unix(),
+						DataKey:   "v",
 					}
 
 					switch item.Value.Value.Type() {
 					case uaGopcua.TypeIDBoolean:
+						msg.DataKey = "vb"
 						msg.Data = item.Value.Value.Bool()
+					case uaGopcua.TypeIDString:
+						msg.DataKey = "vs"
+						msg.Data = item.Value.Value.String()
 					case uaGopcua.TypeIDInt64, uaGopcua.TypeIDInt32, uaGopcua.TypeIDInt16:
-						msg.Data = item.Value.Value.Int()
+						msg.Data = float64(item.Value.Value.Int())
 					case uaGopcua.TypeIDUint64:
-						msg.Data = item.Value.Value.Uint()
+						msg.Data = float64(item.Value.Value.Uint())
 					case uaGopcua.TypeIDFloat, uaGopcua.TypeIDDouble:
 						msg.Data = item.Value.Value.Float()
-					case uaGopcua.TypeIDString:
-						msg.Data = item.Value.Value.String()
 					case uaGopcua.TypeIDByte:
-						msg.Data = item.Value.Value.EncodingMask()
+						msg.DataKey = "vs"
+						msg.Data = string(item.Value.Value.EncodingMask())
+					case uaGopcua.TypeIDDateTime:
+						msg.Data = item.Value.Value.Time()
 					default:
 						msg.Data = 0
 					}
 
 					if err := c.publish(token, msg); err != nil {
-						c.logger.Warn(fmt.Sprintf("failed to publish: %s", err))
+						switch err {
+						case errNotFoundServerURI, errNotFoundNodeID, errNotFoundConn:
+							return err
```

Comment:
I dont understand how is this error returned?
you are waiting on the channel

Reply:
This function runs in a go routine that will stop if there is an error with the routemap